README.md

```diff
@@ -1,2 +1,48 @@
-# TypeScript
+# The Algorithms - TypeScript
+
+<!-- Front Matter -->
 TypeScript Repository of TheAlgorithms, which implements various algorithms and data structures in TypeScript.
+
+<div align="center">
+
+  [![TypeScript Banner][banner]](DIRECTORY.md)
+
+  [![Contributions Welcome][welcome]](CONTRIBUTING.md)
+  [![Discord chat][chat]][discord-server]
+
+  
+</div>
+
+---
+
+<!-- Disclaimer -->
+
+<h4 align="center">
+  These implementations are for demonstrative purposes only. Dedicated implementations of these algorithms and data
+  structures are much better for performance and security reasons. We also do not provide any guarantee for api stability.
+</h4>
+
+---
+
+<!-- Body -->
+
+Before contributing to this repository, make sure to read our [Contribution Guidelines](CONTRIBUTING.md). You can look
+at other [TheAlgorithms Repositories][repositories] or the [issues with a "help wanted" label][help-wanted] for
+inspiration regarding what to implement. Our maintainers will guide you through how to make your contribution properly
+if you make any mistakes. The names of the maintainers of this repository is listed in the
```

Comment:
Minor grammar mistake here.
```suggestion
if you make any mistakes. The names of the maintainers of this repository are listed in the
```


__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix